packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -495,6 +497,62 @@ class _AnimatedPositionedState extends AnimatedWidgetBaseState<AnimatedPositione
   }
 }
 
+/// Animated widget that automatically transitions the child's size over a given
+/// duration whenever the given child's size changes.
+class AnimatedSize extends SingleChildRenderObjectWidget {
+  /// Creates a widget that animates the size of its child implicitly.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  AnimatedSize({
+    Key key,
+    Widget child,
+    this.alignment: FractionalOffset.center,
+    this.curve: Curves.linear,
+    @required this.duration
+  }) : super(key: key, child: child);
+
+  /// How to align the child.
+  ///
+  /// The x and y values of the alignment control the horizontal and vertical
+  /// alignment, respectively.  An x value of 0.0 means that the left edge of
+  /// the child is aligned with the left edge of the parent whereas an x value
+  /// of 1.0 means that the right edge of the child is aligned with the right
+  /// edge of the parent. Other values interpolate (and extrapolate) linearly.
+  /// For example, a value of 0.5 means that the center of the child is aligned
+  /// with the center of the parent.
+  final FractionalOffset alignment;
+
+  /// The curve to apply when animating the parameters of this container.
+  final Curve curve;
+
+  /// The duration over which to animate the parameters of this container.
+  final Duration duration;
+
+  @override
+  RenderAnimatedSize createRenderObject(BuildContext context) {
+    return new RenderAnimatedSize(
+      alignment: alignment,
+      duration: duration,
+      curve: curve,
+      callback: () {
+        new LayoutChangedNotification().dispatch(context);
+      }
```

Comment:
Why does this widget generate layout change notifications but not every other widget?  Maybe we should create a special-purpose widget that's entire job is to generate these layout change notifications rather than adding that functionality to every other widget.